cardano-node-emulator/src/Cardano/Node/Emulator/API.hs

```diff
@@ -152,6 +161,16 @@ nextSlot = do
 currentSlot :: (MonadEmulator m) => m Slot
 currentSlot = handleChain E.getCurrentSlot
 
+-- | Get the time range of the current slot of the emulated node.
+currentTimeRange :: (MonadEmulator m) => m (POSIXTime, POSIXTime)
+currentTimeRange = do
+  slotConfig <- asks pSlotConfig
+  slot <- currentSlot
+  pure
+    ( slotToBeginPOSIXTime slotConfig slot
+    , slotToEndPOSIXTime slotConfig slot
+    )
+
 -- | Call `nextSlot` until the current slot number equals or exceeds the given slot number.
```

Comment:
Not among your changes for this PR, so feel free to ignore this. 

I noticed the comment is slightly inaccurate: `nextSlot` will be called until the current slot number *exceeds* the given slot number. If they are equal, `pure ()` is called.